src/main/webapp/app/overview/submission-result-status.component.ts

```diff
@@ -48,7 +49,23 @@ export class SubmissionResultStatusComponent implements OnChanges {
             this.quizNotStarted = ArtemisQuizService.notStarted(quizExercise);
         } else {
             this.uninitialized = !afterDueDate && !this.studentParticipation;
-            this.notSubmitted = !!this.studentParticipation && !this.studentParticipation.submissions?.length;
+            this.notSubmitted = afterDueDate && !!this.studentParticipation && !this.studentParticipation.submissions?.length;
+        }
+
+        this.setShouldShowResult(afterDueDate);
+    }
+
+    private setShouldShowResult(afterDueDate: boolean) {
+        if (this.exercise.type === ExerciseType.QUIZ) {
+            this.shouldShowResult = !!this.studentParticipation?.results?.length;
+        } else if (this.exercise.type === ExerciseType.PROGRAMMING) {
+            const initializationStatesForResult = [InitializationState.INITIALIZED, InitializationState.INACTIVE, InitializationState.FINISHED];
+            this.shouldShowResult =
+                (!!this.studentParticipation?.results?.length || !afterDueDate) &&
```

Comment:
`!!this.studentParticipation?.results?.length` and `this.studentParticipation?.initializationState` could be variables in this method. This would make those conditions easier to read. For example:
```ts
const anyResultPresent = !!this.studentParticipation?.results?.length;
```
or event shorter (should show the same thing, right?):
```ts
const anyResultPresent = !!this.studentParticipation?.results;
```

Reply:
You cannot leave out the length, since this would change the behavior for an empty array.
At least for me the optional chaining is also quite readable is it about that or that is is repeated for the quiz exercise as well? I'd find it more complicated if I have an additional variable outside the if that is then used in some but not all branches.